distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso

```diff
@@ -12,103 +12,199 @@ import project.Data.Sort_Column.Sort_Column
+        selection = self.select_columns_helper selector reorder=True problem_builder
+        problem_builder.attach_problems_before on_problems <|
+            selected_names = Map.from_vector (selection.map column-> [column.name, True])
+            other_columns = self.internal_columns.filter column->
+                is_selected = selected_names.get_or_else column.name False
+                is_selected.not
+            result = case position of
+                Position.Before_Other_Columns -> selection + other_columns
+                Position.After_Other_Columns -> other_columns + selection
+            result
+
+    ## PRIVATE
+       A helper function which selects columns from the table based on the
+       provided selection criteria.
+
+       Arguments:
+       - selector: Column selection criteria or vector of column names.
+       - reorder: Specifies whether to reorder the matched columns according to
+         the order of the selection criteria.
+         If `False`, the matched entries are returned in the same order as in
+         the input.
+         If `True`, the matched entries are returned in the order of the
+         criteria matching them. If a single object has been matched by multiple
+         criteria, it is placed in the group belonging to the first matching
+         criterion on the list. If a single criterion's group has more than one
+         element, their relative order is the same as in the input.
+       - problem_builder: Encapsulates the aggregation of encountered problems.
+    select_columns_helper : Vector | Column_Selector -> Boolean -> Problem_Builder -> Vector
+    select_columns_helper self selector reorder problem_builder = case selector of
+        _ : Vector.Vector ->
+            self.select_columns_helper (Column_Selector.By_Name selector) reorder problem_builder
+        Column_Selector.By_Name names matcher ->
+            valid_names = validate_unique names problem_builder.report_duplicate_column_selectors
+            Matching.match_criteria_callback matcher self.internal_columns valid_names reorder=reorder name_mapper=(_.name) problem_callback=problem_builder.report_missing_input_columns
+        Column_Selector.By_Index indices ->
+            good_indices = validate_indices self.internal_columns.length indices problem_builder . map .first
+            case reorder of
+                True ->
+                    select_indices_reordering self.internal_columns good_indices
+                False ->
+                    select_indices_preserving_order self.internal_columns good_indices
+        Column_Selector.By_Column columns ->
+            column_names = columns.map .name
+            new_selector = Column_Selector.By_Name column_names Text_Matcher.Case_Sensitive
+            self.select_columns_helper new_selector reorder=reorder problem_builder=problem_builder
+        Column_Selector.Blank_Columns when_any treat_nans_as_blank -> if self.internal_columns.is_empty then [] else
```

Comment:
This is the relevant part, the rest of this file is mostly moving functions around - I moved common functions relying on `select_columns_helper` to the `Table_Column_Helper` so that we can create the helper type in one place in each table, encapsulating any changes to the common logic that is required for selecting columns (now we needed to add additional callbacks to make it possible to select blank columns, it may change again in the future so I want to amend only `Table.columns_helper` instead of all `Table_Helpers` invocations within Table).